File: frontend/src/infrastructure/ai/prompts/queryPrompt.ts
import type { AIClient } from '../provider';

const QUERY_SYSTEM_PROMPT = `You are a natural language to filter converter for a plastic recycling platform.
Convert the user's question into a JSON filter object with optional fields:
- materialType: PET, HDPE, PP, LDPE, PVC, mixed
- stage: collection, sorting, processing, output, dispatch
- vendor: string
- startDate: ISO date
- endDate: ISO date
- aggregation: total, average, count
Return ONLY valid JSON.`;

export async function buildQueryFilters(
  ai: AIClient,
  message: string,
): Promise<Record<string, unknown>> {
  const raw = await ai.complete(message, QUERY_SYSTEM_PROMPT);
  try {
    return JSON.parse(raw);
  } catch {
    return {};
  }
}

File: backend/traceflow-api/src/infrastructure/adapters/ai/prompts/query_prompt.py
QUERY_SYSTEM_PROMPT = """You are a query parser for TraceFlow, a plastic recycling supply chain system.

Given a natural language question about supply chain data, extract structured query filters as JSON.

Fields to extract:
- "metric": What is being asked about ("total", "dispatched", "processed", "loss", "average_loss", "count")
- "date_range_start": Start date (ISO format or relative like "7 days ago")
- "date_range_end": End date (ISO format or relative like "today")
- "stage": Processing stage filter (sorting, washing, shredding, extrusion, pelletizing, dispatch)
- "material": Material type filter (PET, HDPE, PP, LDPE, PVC, mixed)
- "vendor": Vendor name filter
- "batch_id": Specific batch ID filter

Only include fields that are mentioned or implied in the question.

Respond ONLY with valid JSON. Example:
Input: "How much PET was dispatched last week?"
Output: {"metric": "dispatched", "material": "PET", "date_range_start": "7 days ago", "date_range_end": "today"}
"""
